cms/djangoapps/contentstore/views/course.py

```diff
@@ -283,6 +293,41 @@ def course_rerun_handler(request, course_key_string):
             })
 
 
+@login_required
+@ensure_csrf_cookie
+@require_http_methods(["GET"])
+def course_index_handler(request, course_key_string):
+    """
+    The restful handler for course indexing.
+    GET
+        json: return status of indexing task
+    """
+    # Only global staff (PMs) are able to rerun courses during the soft launch
+    if not GlobalStaff().has_user(request.user):
+        raise PermissionDenied()
+    course_key = CourseKey.from_string(course_key_string)
+    with modulestore().bulk_operations(course_key):
+        if request.method == 'GET':
+            error = reindex_course_and_check_access(course_key, request.user, depth=3)
+            if error:
+                return HttpResponse(
```

Comment:
Why not use the [`JsonResponse`](https://github.com/edx/edx-platform/blob/master/common/djangoapps/util/json_request.py#L32) class? Also, if this is an error response, it must use an appropriate [HTTP status code](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes), not the default `200 OK`.